runtime/vm/ContinuationHelpers.cpp

```diff
@@ -923,14 +923,110 @@ preparePinnedVirtualThreadForUnmount(J9VMThread *currentThread, j9object_t syncO
 	UDATA result = J9_OBJECT_MONITOR_YIELD_VIRTUAL;
 	J9ObjectMonitor *syncObjectMonitor = NULL;
 	J9ObjectMonitor *enteredMonitorsList = NULL;
-	j9objectmonitor_t lock = 0;
-	j9object_t continuationObj = NULL;
 	UDATA monitorCount = 0;
 	J9JavaVM *vm = currentThread->javaVM;
 
 	if (NULL != syncObj) {
+		j9objectmonitor_t volatile *lwEA = VM_ObjectMonitor::inlineGetLockAddress(currentThread, syncObj);
```

Comment:
Should hooks like `J9HOOK_VM_MONITOR_CONTENDED_ENTER` be triggered somewhere in this function ?

Reply:
Will create a separate PR to fix the jvmti contended enter hook. That will require some additional changes to store the monitor state when yielding.